### PR TITLE
Docs - add sections on configuring page templates & key terms

### PR DESCRIPTION
**Pull request by emcarru**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
...
+
+For example, take `templates/page.php`:
+
+```php
+<?php
+
+Flynt\echoHtmlFromConfigFile('default.json');
+```
+
+The JSON template configuration files are found in `config/templates`. These configuration files define the [areas](#areas) and [components](#components) which are loaded into the template.
```
Comment (domtra) [inline README.md]: i'd rather say `files define the components and their areas which are loaded`

File: README.md [hunk 2]
```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
...
+
+Take `config/templates/default.json` as an example. This template calls the `DocumentDefault` component and registers one area: `layout`. The `layout` area calls the `LayoutSinglePost` component, which in turn contains three nested [areas](#areas): `mainHeader`, `pageComponents`, and `mainFooter`. In addition, the `pageComponents` area calls the `ComponentLoaderFlexible` component.
```
Comment (domtra) [inline README.md]: i am not really happy with `calls` and `registers`. components are registered somewhere else. and not called. What about contains, consists of, is made up of / is composed with?. I am sure you have some ideas ;)

File: README.md [hunk 3]
```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
...
+          ],
+          "mainFooter": []
+        }
+      }
+    ]
+  }
+}
+```
+
+To complete the picture, the `layout` area defined in the template is then output in `Components/DocumentDefault/index.twig`:
```
Comment (domtra) [inline README.md]: this sentence confuses me

File: README.md [hunk 4]
```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
...
+
+Building components is a sustainable process, meaning every component you develop can be reused within a project, or in another; increasing your head-start with every new Flynt project.
+
+### Areas
+Since components are self-contained, areas provide a way to stack our building-blocks together. An area is simply a location within a component where it is possible to add other components.
+
+### Features
+With WordPress, it is easy to create one large `functions.php` file, crammed full of all the custom logic your theme may need. This can get messy. In Flynt, we split each piece of functionality into smaller, self-contained **feature** bundles.
+
+In most cases, features add global hooks and filters that affect the project on a global level. With this in mind, each feature is built with drag and drop reusability in mind.
```
Comment (domtra) [inline README.md]: `drag & drop reusability` reminds me of backend d&d feature. just `reusability`?

**Comment by emcarru** [comment]
@domtra: Updated, thanks for the feedback. I've removed `to complete the picture` since that wasn't really adding anything anyway. Though I think it still makes more sense to keep `take ... as an example`, since this is the usage section, and we are trying to explain the concept/usage through an example.

**Pull request merged**